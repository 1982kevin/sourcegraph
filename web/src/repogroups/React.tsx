import { RepogroupMetadata } from './types'
import { SearchPatternType } from '../../../shared/src/graphql/schema'
import * as React from 'react'
export const react: RepogroupMetadata = {
    title: 'React',
    name: 'react',
    url: '/react',
    description:
        "Explore React repositories on GitHub. The search examples show how to use React Hook useState with various data types.",
    examples: [
        {
            title: 'Find imports of useState with regex search',
            exampleQuery: <>import [^;]+useState[^;]+ from 'react'</>,
            rawQuery: "import [^;]+useState[^;]+ from 'react'",
            patternType: SearchPatternType.regexp,
        },
        {
            title: 'Examples of useState with an object as the input parameter',
            description: `useState takes a single argument and can accept a primitive, an array or an object.
            Syntax for usage is [state_value, function_to_update_state_value] = useState (initial_state_value).`,
            exampleQuery: <>{'useState({:[string]})'}</>,
            rawQuery: 'useState({:[string]}) count:1000',
            patternType: SearchPatternType.structural,
        },
        {
            title: 'Examples of useState with any JavaScript data type as the input parameter',
            exampleQuery: <>useState(:[string])</>,
            rawQuery: 'useState(:[string]) count:1000',
            patternType: SearchPatternType.structural,
        },
        {
            title: 'Examples of useState with any JavaScript data type as the input parameter in only TypeScript files',
            exampleQuery: (
                <>
                    useState(:[string])
                    <span className="search-keyword">lang:</span>typescript
                </>
            ),
            rawQuery: 'useState(:[string]) count:1000 lang:typescript',
            patternType: SearchPatternType.structural,
        },
    ],
    homepageDescription: 'Learn React with code examples.',
    homepageIcon: 'https://cdn4.iconfinder.com/data/icons/logos-3/600/React.js_logo-512.png',
}
