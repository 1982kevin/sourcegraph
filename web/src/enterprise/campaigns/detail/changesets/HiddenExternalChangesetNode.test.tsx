import React from 'react'
import { createRenderer } from 'react-test-renderer/shallow'
import { HiddenExternalChangesetNode } from './HiddenExternalChangesetNode'
import { addDays } from 'date-fns'
import { ChangesetState } from '../../../../../../shared/src/graphql/schema'

describe('HiddenExternalChangesetNode', () => {
    test('renders', () => {
        const renderer = createRenderer()
        renderer.render(
            <HiddenExternalChangesetNode
                node={{
                    id: 'test',
                    nextSyncAt: addDays(new Date(), 1).toISOString(),
                    state: ChangesetState.OPEN,
                    updatedAt: new Date().toISOString(),
                }}
            />
        )
        const result = renderer.getRenderOutput()
        expect(result.props).toMatchSnapshot()
    })
})
