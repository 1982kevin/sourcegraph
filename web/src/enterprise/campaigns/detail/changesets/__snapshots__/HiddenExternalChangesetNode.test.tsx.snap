// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HiddenExternalChangesetNode renders 1`] = `
Object {
  "children": <div
    className="changeset-node__content changeset-node__content--no-collapse flex-fill"
  >
    <div
      className="d-flex align-items-start m-1 ml-2"
    >
      <div
        className="changeset-node__content flex-fill"
      >
        <div
          className="d-flex flex-column"
        >
          <div
            className="m-0 mb-2"
          >
            <h3
              className="m-0 d-inline"
            >
              <ChangesetStateIcon
                state="OPEN"
              />
              <span
                className="text-muted"
              >
                Changeset in a private repository
              </span>
            </h3>
          </div>
          <div>
            <ChangesetLastSynced
              changeset={
                Object {
                  "id": "test",
                  "nextSyncAt": "2006-01-03T22:04:05.000Z",
                  "state": "OPEN",
                  "updatedAt": "2006-01-02T22:04:05.000Z",
                }
              }
              viewerCanAdminister={false}
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "className": "list-group-item e2e-changeset-node",
}
`;
